chronos/asyncfutures2.nim

```diff
@@ -931,13 +946,13 @@ proc oneValue*[T](futs: varargs[Future[T]]): Future[T] {.
 
   return retFuture
 
-proc cancelAndWait*[T](fut: Future[T]): Future[void] =
+proc cancelAndWait*[T](fut: Future[T]): RaiseTrackingFuture[void, (CancelledError,)] =
```

Comment:
```suggestion
proc cancelAndWait*[T](fut: Future[T]): Future[void] {.asyncraises:[(CancelledError,)] =
```
the rewrite to RTF should be macro-handled, no?

Reply:
We don't have access to macros here (asyncmacro2 requires asyncfutures2)

Reply:
maybe we should move this one elsewhere then - ie `cancelAndWait` is a composite "utility" operation much like `allFutures` etc - then we keep "primitives" in asyncfuture2 and maybe one fine day we can get rid of include altogether